core/api/src/app/support/chat.ts

```diff
@@ -1,20 +1,63 @@
+import { Users } from "@/app"
 import { RepositoryError, CouldNotFindError } from "@/domain/errors"
+import { ChatAssistantNotFoundError } from "@/domain/support/errors"
 import { AccountsRepository, UsersRepository } from "@/services/mongoose"
 import { SupportChatRepository } from "@/services/mongoose/support-chat"
 import { Assistant } from "@/services/openai"
 
+const getMessages = async ({
+  supportChatId,
+  accountId,
+}: {
+  supportChatId: SupportChatId
+  accountId: AccountId
+}) => {
+  const messages = Assistant().getMessages(supportChatId)
+  if (messages instanceof ChatAssistantNotFoundError) {
+    await initializeSupportChat({ accountId })
+    return []
+  }
+
+  return messages
+}
+
+export const initializeSupportChat = async ({ accountId }: { accountId: AccountId }) => {
+  const account = await AccountsRepository().findById(accountId)
+  if (account instanceof RepositoryError) return account
+
+  const user = await UsersRepository().findById(account.kratosUserId)
+  if (user instanceof RepositoryError) return user
+
+  const countryCode = user.phoneMetadata?.countryCode ?? "unknown"
+  const level = account.level
+  const language = await Users.getUserLanguage(user.id)
+
+  const supportChatId = await Assistant().initialize({
+    level,
+    countryCode,
+    language,
+  })
+  if (supportChatId instanceof Error) return supportChatId
+
+  const res = await SupportChatRepository().create({ supportChatId, accountId })
+  if (res instanceof RepositoryError) return res
+
+  return supportChatId
+}
+
 export const getSupportChatMessages = async (accountId: AccountId) => {
   const supportChatId = await SupportChatRepository().findNewestByAccountId(accountId)
 
   if (supportChatId instanceof CouldNotFindError) {
     return []
-  }
-
-  if (supportChatId instanceof Error) {
+  } else if (supportChatId instanceof Error) {
```

Comment:
not sure to understand this change... else block is not necessary

Reply:
it is, there is a different between the prior block which is checking `CouldNotFindError` and this block checking for the more general `Error`

Reply:
again not undestanding.. not a blocker but the else is unnecessary because you are using return